tests/Unit/ParallelAlgorithms/Events/Test_ObserveNorms.cpp

```diff
@@ -304,6 +314,130 @@ void test(const std::unique_ptr<ObserveEvent> observe,
+                     Spectral::Quadrature::GaussLobatto,
+                     Spectral::Quadrature::AxialSymmetry}}};
+    const domain::CoordinateMap<
+        Frame::ElementLogical, Frame::Inertial,
+        domain::CoordinateMaps::ProductOf3Maps<Affine, Affine, Identity1D>>
+        map{{Affine{-1.0, 1.0, 1.0, 2.0}, Affine{-1.0, 1.0, 2.0, 3.0},
+             Identity1D{}}};
+    inertial_coords = map(logical_coordinates(mesh));
+    det_inv_jacobian =
+        determinant(map.inv_jacobian(logical_coordinates((mesh))));
+    expected_volume = 3.0 * M_PI;
+  }
+  const size_t num_points = mesh.number_of_grid_points();
+
+  const double observation_time = 2.0;
+  Variables<tmpl::list<Var0, Var1>> vars(num_points);
+
+  auto& scalar = get<Var0>(vars);
+  get<>(scalar) = get<0>(inertial_coords);
+
+  get<Var1>(vars) = inertial_coords;
+
+  ActionTesting::MockRuntimeSystem<metavariables> runner{{}};
+  ActionTesting::emplace_component<element_component>(make_not_null(&runner),
+                                                      array_index);
+  ActionTesting::emplace_group_component<observer_component>(&runner);
+
+  auto box = db::create<
+      db::AddSimpleTags<Parallel::Tags::MetavariablesImpl<metavariables>,
+                        ::Events::Tags::ObserverMesh<3>,
+                        ::Events::Tags::ObserverDetInvJacobian<
+                            Frame::ElementLogical, Frame::Inertial>,
+                        domain::Tags::Coordinates<3, Frame::Inertial>,
+                        Tags::Variables<typename decltype(vars)::tags_list>,
+                        observers::Tags::ObservationKey<ArraySectionIdTag>>>(
+      metavariables{}, mesh, det_inv_jacobian, inertial_coords, vars, section);
+
+  auto obs_box = make_observation_box<
+      tmpl::filter<typename ObserveNormsEvent<
+                       ArraySectionIdTag>::compute_tags_for_observation_box,
+                   db::is_compute_tag<tmpl::_1>>>(make_not_null(&box));
+  observe->run(make_not_null(&obs_box),
+               ActionTesting::cache<element_component>(runner, array_index),
+               array_index, std::add_pointer_t<element_component>{},
+               {"TimeName", observation_time});
+
+  // Process the data
+  runner.template invoke_queued_simple_action<observer_component>(0);
+  CHECK(runner.template is_simple_action_queue_empty<observer_component>(0));
+
+  const auto& results = MockContributeReductionData::results;
+
+  CHECK(results.reduction_names[2] == "Volume");
+  CHECK(results.volume == approx(expected_volume));
+
+  // Check L2 integral norms
+  CHECK(results.reduction_names[3] == "L2IntegralNorm(Var1)");
+  CHECK(results.reduction_names[4] == "L2IntegralNorm(Var1_x)");
+  CHECK(results.reduction_names[5] == "L2IntegralNorm(Var1_y)");
+  CHECK(results.reduction_names[6] == "L2IntegralNorm(Var1_z)");
+  if constexpr (Spherical) {
+    CHECK(results.l2_integral_norm_values[0] == approx(0.79056941504209488));
```

Comment:
where do these numbers come from?  I don't like magic numbers, can they be computed analytically?